lib/client/src/gpa.rs

```diff
@@ -148,3 +149,44 @@ pub async fn fetch_multiple_accounts(
         .map(|(acc, key)| (*key, acc.unwrap().into()))
         .collect())
 }
+
+pub async fn fetch_multiple_accounts_in_chunks(
+    rpc: &RpcClientAsync,
+    keys: &[Pubkey],
+    max_chunk_size: usize,
+    parallel_rpc_requests: usize,
+) -> anyhow::Result<Vec<(Pubkey, Account)>> {
+    let config = RpcAccountInfoConfig {
+        encoding: Some(UiAccountEncoding::Base64),
+        ..RpcAccountInfoConfig::default()
+    };
+
+    let mut raw_result = stream::iter(keys)
+        .chunks(max_chunk_size)
+        .map(|keys| {
+            let account_info_config = config.clone();
+            async move {
+                let mut keys = keys.iter().map(|x| **x).collect::<Vec<Pubkey>>();
+                let req_res = rpc
+                    .get_multiple_accounts_with_config(&keys, account_info_config)
+                    .await;
+
+                match req_res {
+                    Ok(v) => Ok(keys.drain(..).zip(v.value).collect::<Vec<_>>()),
+                    Err(e) => Err(e),
+                }
+            }
+        })
+        .buffer_unordered(parallel_rpc_requests)
+        .collect::<Vec<_>>()
+        .await;
+
+    let result = raw_result
+        .drain(..)
+        .map(|v| v.unwrap())
+        .flatten()
+        .filter_map(|x| x.1.map(|o| (x.0, o)))
```

Comment:
prefer: `.filter_map(|(pubkey, account_opt)| account_opt.map(|acc| (pubkey, acc)))`  (but even that needs some thought to read, unfortunately)

Is it even acceptable to not return data for some of the accounts that were requested? Could you put in the doc comment that this is could return no data without an error?

Reply:
I'd appreciate the comment